src/index.js

```diff
@@ -125,8 +125,9 @@ export let ContextTree = _.curry(
       prepForUpdate(tree)
       try {
         await processResponse(await service(dto, now))
-      } catch (e) {
-        onError(e)
+      } catch (error) {
+        onError(error) // Raise the onError event
+        await processResponse(tree)
```

Comment:
This will only happen if you don't `throw` in a custom onError, which isn't the default behavior. This means that this won't get called by default which doesn't seem obvious. Maybe we should move this up a line?

Reply:
I'm definitely open to discussion on this 🤔